### PR TITLE
buildbot: Update to 2.8.3

### PR DESCRIPTION
**Pull request by ryandesign**
#### Description

This PR updates the buildbot ports to 2.8.3.

I've reorganized each Portfile slightly, removed unnecessary overrides of default python portgroup behavior, used https for homepages, adjusted descriptions.

I've added the py-buildbot-www dependency to py-buildbot-console-view, py-buildbot-grid-view, and py-buildbot-waterfall, since those are all web UI plugins.

I've removed the py-buildbot and py-mock dependencies from py-buildbot-www per https://github.com/buildbot/buildbot/commit/26af545b3f8cbf97f2335e61b08dcf30f447c17b.

I've combined the py-buildbot and buildbot-2 ports and renamed it to buildbot. I've added console, grid, waterfall and www variants to let the user choose which of those they want to include; they're all enabled by default. I've added an upstream patch to fix a performance issue.

py-buildbot-macports-custom-views is updated to 0.1.2.

###### Tested on
macOS 10.13.6 17G12034
Xcode 9.4.1 9F2000

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] tried a full install with `sudo port -vst install`?

### CODE REVIEW

File: devel/buildbot-worker/Portfile [hunk 1]
```diff
@@ -4,25 +4,26 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                buildbot-worker
-version             2.8.2
+version             2.8.3
```
Comment (mojca) [inline devel/buildbot-worker/Portfile:7]: 2.8.4 has just been released

File: python/py-buildbot-macports-custom-views/Portfile [hunk 2]
```diff
@@ -4,31 +4,30 @@ PortSystem          1.0
...
-homepage            https://github.com/macports/buildbot-macports-custom-views
-master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
-distname            ${python.rootname}-${version}
+description         MacPorts custom views plugin for Buildbot web UI
+long_description    {*}${description}
 
-checksums           sha256  4fcc176f033f5b838dd7376628ef2054053f3b8f51560940e510e75edea23a5c \
-                    rmd160  cd224aaa98114324b046094f6d7b2aac8b31389a \
-                    size    447418
+homepage            https://github.com/macports/buildbot-macports-custom-views
```
Comment (mojca) [inline python/py-buildbot-macports-custom-views/Portfile:21]: Please note that most of these newer ports have likely been generated with upt. If you feel that defaults from upt could be improved, please suggest changes. @reneeotten

**Review by mojca — COMMENTED** [review]
Looks reasonable.

File: devel/buildbot-worker/Portfile (hunk 1 above)
Comment (ryandesign) [reply devel/buildbot-worker/Portfile:7]: I can work on that next.

File: python/py-buildbot-macports-custom-views/Portfile (hunk 2 above)
Comment (ryandesign) [reply python/py-buildbot-macports-custom-views/Portfile:21]: I've never looked at upt. I can't look at everything. :)

**Pull request merged**